school.py

```diff
@@ -0,0 +1,87 @@
+import math
+import random
+import xml.etree.ElementTree as ET
+import sumolib
+import logging
+
+from typing import Tuple
+from perlin import POPULATION_BASE, get_population_number
+from utility import find_city_centre, radius_of_network, k_means_clusters
+
+
+def find_school_edges(net: sumolib.net.Net, num_schools: int, centre: Tuple[float, float]):
+    # Use k-means, to split the net into num_schools number of clusters, each containing approx same number of edges
+    districts = k_means_clusters(net, num_schools)
+
+    school_edges = []
+    centre = find_city_centre(net)
+    radius = radius_of_network(net, centre)
+    # Sort each edge in each district based on their noise, and return edge with highest noise from each district
+    for district in districts:
+        district.sort(key=lambda x: get_population_number(x, centre=centre, radius=radius, base=POPULATION_BASE))
+        school_edges.append(district[-1])
+
+    return school_edges
+
+
+def setup_schools(args, net: sumolib.net.Net, stats: ET.ElementTree, school_count: int or None,
+                  centre: Tuple[float, float]):
+    xml_schools = stats.find('schools')
+    if xml_schools is None:
+        xml_schools = ET.SubElement(stats.getroot(), "schools")
+    if school_count is None:
+        # Voodoo parameter, seems to be about the value for a couple of danish cities.
+        # In general one high school, per 5000-7000 inhabitant in a city, so 0.2 pr 1000 inhabitants
+        schools_per_1000_inhabitants = float(args["--schools.ratio"])
+
+        # Calculate default number of schools, based on population if none input parameter
+        xml_general = stats.find('general')
+        inhabitants = xml_general.get('inhabitants')
+        num_schools_default = math.ceil(int(inhabitants) * schools_per_1000_inhabitants / 1000)
+
+        # Number of new schools to be placed
+        number_new_schools = num_schools_default - len(xml_schools.findall("school"))
+    else:
+        # Else place new number of schools as according to input
+        number_new_schools = school_count - len(xml_schools.findall("school"))
+
+    if number_new_schools == 0:
+        return
+    if number_new_schools < 0:
+        print(f"Warning: {school_count} schools was requested, but there are already {len(xml_schools)} defined")
```

Comment:
We should use logging now. Remember to import it.

```suggestion
        logging.warning(f"{school_count} schools was requested, but there are already {len(xml_schools)} defined")
```

Reply:
_it is imported, disregard_

Reply:
Committed in 48d911155f995f23c7a6848ff9bf09c9969c200d